### DISCUSSION TITLE
Using "constructor" argument in genesis alloc

### DISCUSSION THREAD

**Discussion by debjit-bw** [paradigmxyz/reth#13413] [Technical discussions]
I want to contribute to a feature: parsing the `constructor` argument in the genesis alloc, apart from `code` in order to initialise the genesis state. This would make it easier to use allocs designed for other clients like [Nethermind](https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Chains/chiado.json#L255C7-L255C20) or [Erigon](https://github.com/erigontech/erigon/blob/main/core/allocs/chiado.json#L50C5-L50C18).

Can you give me an idea how to proceed?

**Comment by mattsse** [comment]
I think for this we'd need a custom serde for this?

https://github.com/alloy-rs/alloy/blob/28ad3f778d44c743408963b244e04b458591b160/crates/genesis/src/lib.rs#L46

**Reply by debjit-bw** [reply]
yes, but in addition to deserializing, we also need to make other changes to accounts. for example, some constructors modify the storage roots of some other accounts, so need the evm while parsing. i'm not sure how to include those changes.
